src/protocol/payload/mod.rs

```diff
@@ -121,11 +123,11 @@ impl VarInt {
 struct VarStr(String);
 
 impl VarStr {
-    fn encode(&self, buffer: &mut Vec<u8>) -> io::Result<usize> {
-        let int_len = VarInt(self.0.len()).encode(buffer)?;
```

Comment:
Note the removal of `<usize>` -- required to meet the Trait definitation, and I couldn't find a use case for this (?).